OOStepByStep/ClassRoom.cs

```diff
@@ -0,0 +1,58 @@
+﻿using OOStepByStep.Human;
+using System;
+using System.Collections.Generic;
+using System.Linq;
+using System.Text;
+using System.Threading.Tasks;
+
+namespace OOStepByStep
+{
+    public class ClassRoom
+    {
+        private int classNum;
+        private Teacher teacher = null;
+        private List<Student> students = new ();
+
+        public ClassRoom(int classNum)
+        {
+            this.classNum = classNum;
+        }
+
+        public Dictionary<string, List<string>> Join(Student student)
```

Comment:
Probably you could merge the Join method into one by using Person as input type and using xx.GetType() == typeof(Teacher).